src/psyclone/transformations.py

```diff
@@ -1252,6 +1251,148 @@ def apply(self, target_nodes, options=None):
+        self._omp_nowait = value
+
+    def _directive(self, children):
+        '''
+        Creates the type of directive needed for this sub-class of
+        transformation.
+        :param children: list of Nodes that will be the children of \
+                         the created directive.
+        :type children: list of :py:class:`psyclone.psyir.nodes.Node`
+        :rtype: :py:class:`psyclone.psyGen.OMPDoDirective`
+        :raises NotImplementedError: if a collapse argument is supplied
+        '''
+        _directive = nodes.OMPSingleDirective(children=children,
+                                        nowait=self.omp_nowait)
+        return _directive
+
+    def validate(self, node_list, options=None):
+        '''
+        Perform OpenMP-specific validation checks.
+
+        :param node_list: list of Nodes to put within serial region.
+        :type node_list: list of :py:class:`psyclone.psyir.nodes.Node`
+        :param options: a dictionary with options for transformations.
+        :type options: dictionary of string:values or None
+        :param bool options["node-type-check"]: this flag controls if the \
+                type of the nodes enclosed in the region should be tested \
+                to avoid using unsupported nodes inside a region.
+
+        :raises TransformationError: if the target Nodes are already within \
+                                     some OMP single region.
+        '''
+
+        if node_list[0].ancestor(nodes.OMPSingleDirective):
+            raise TransformationError("Error in OMPSingle transformation:" +
+                                      " cannot create an OpenMP SINGLE " +
+                                      "region within another OpenMP SINGLE region.")
+
+        # Now call the general validation checks
+        super(OMPSingleTrans, self).validate(node_list, options)
+
+    def apply(self, node, options=None):
+        '''Apply the OMPSingleTrans transformation to the specified node in a
+        Schedule. 
+
+        At code-generation time (when
+        :py:meth:`OMPLoopDirective.gen_code` is called), this node must be
+        within (i.e. a child of) an OpenMP PARALLEL region.
+
+        If the keyword "nowait" is specified in the options, it will cause a
+        nowait clause to be added if it is set to True, otherwise
+        the no nowait clause will be added.
+
+        :param node: the supplied node to which we will apply the \
+                     OMPLoopTrans transformation
+        :type node: :py:class:`psyclone.psyir.nodes.Node`
+        :param options: a dictionary with options for transformations\
+                        and validation.
+        :type options: dictionary of string:values or None
+        :param bool options["nowait"]:
+                indicating whether or not to use a nowait clause on this single \
+                region.
+
+        :returns: (:py:class:`psyclone.psyir.nodes.Schedule`, \
```

Comment:
No need for a line-continuation if using parentheses. In fact, our new way of doing things is not to return anything at all so please could you adjust accordingly.

Reply:
As in the `apply` function should no longer return a modified schedule and a record of the transformation? 
I adjusted to match that for now but I can remove if thats what you mean.

Reply:
Yes, it should no longer return a modified schedule and a record. However, if the parent class still does so (and it probably does) then what you have is fine.

Reply:
Yeah, I've checked after rebasing and the latest master still has the parent class doing so